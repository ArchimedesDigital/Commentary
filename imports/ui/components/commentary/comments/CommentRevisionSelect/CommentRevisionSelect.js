import React from 'react';
import FlatButton from 'material-ui/FlatButton';
import CommentCitation from '/imports/ui/components/commentary/comments/CommentCitation';

import moment from 'moment';

/*
	helpers
*/
const getDateRevision = (revision) => {
	if (revision.originalDate) return revision.originalDate;
	else if (revision.updated) return revision.updated;
	return revision.created;
};

const getClassName = (selectedRevisionIndex, i) => (`revision ${selectedRevisionIndex === i ? 'selected-revision' : ''}`);

const sortRevisions = (revisions) => (_.sortBy(revisions, 'created'));


/*
	BEGIN CommentRevisionSelect
*/
const CommentRevisionSelect = props => (
	<div className="comment-revisions">

		{!props.isTest && sortRevisions(props.comment.revisions).map((revision, i) => {

			const updated = getDateRevision(revision);

			return (
				<FlatButton
					key={revision._id}
					id={i}
					data-id={revision._id}
					className={getClassName(props.selectedRevisionIndex, i)}
					onClick={props.selectRevision}
					label={`Revision ${moment(updated).format('D MMMM YYYY')}`}
				/>
			);
		})}
		<CommentCitation
			comment={props.comment}
		/>
	</div>
);
CommentRevisionSelect.propTypes = {
	comment: React.PropTypes.shape({
		_id: React.PropTypes.string.isRequired,
		revisions: React.PropTypes.arrayOf(React.PropTypes.shape({
			_id: React.PropTypes.string.isRequired,
			created: React.PropTypes.instanceOf(Date),
			updated: React.PropTypes.instanceOf(Date),
			originalDate: React.PropTypes.instanceOf(Date),
		})).isRequired,
		urn: React.PropTypes.string.isRequired,
	}),
	selectedRevisionIndex: React.PropTypes.number.isRequired,
	selectRevision: React.PropTypes.func.isRequired,
	isTest: React.PropTypes.bool,
};
CommentRevisionSelect.defaultProps = {

};
/*
	END CommentRevisionSelect
*/

export default CommentRevisionSelect;
