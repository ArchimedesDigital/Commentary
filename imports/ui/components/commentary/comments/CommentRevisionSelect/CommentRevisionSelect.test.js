import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import CommentRevisionSelect from './CommentRevisionSelect';

describe('CommentRevisionSelect', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<CommentRevisionSelect
						commentId="testId"
						revisions={[
							{_id: 'testId'},
						]}
						selectedRevisionIndex={0}
						selectRevision={() => {}}
					/>
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
