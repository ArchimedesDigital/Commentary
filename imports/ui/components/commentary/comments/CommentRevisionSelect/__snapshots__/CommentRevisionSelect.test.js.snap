// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentRevisionSelect renders correctly 1`] = `
<div
  className="comment-revisions"
>
  <button
    className="revision selected-revision"
    data-id="testId"
    disabled={false}
    href={undefined}
    id={0}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "rgba(0, 0, 0, 0)",
        "border": 10,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 36,
        "lineHeight": "36px",
        "margin": 0,
        "minWidth": 88,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
        "verticalAlign": null,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <span
        style={
          Object {
            "fontSize": 14,
            "fontWeight": 500,
            "letterSpacing": 0,
            "muiPrepared": true,
            "paddingLeft": 16,
            "paddingRight": 16,
            "position": "relative",
            "textTransform": "uppercase",
            "verticalAlign": "middle",
          }
        }
      >
        Revision 22 May 2017
      </span>
    </div>
  </button>
  <div
    className="comment-citation"
  >
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "2px",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "minWidth": 88,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "#ffffff",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "margin": 0,
                  "muiPrepared": true,
                  "opacity": 1,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "userSelect": "none",
                }
              }
            >
              Cite this comment
            </span>
          </div>
        </div>
      </button>
    </div>
    <div
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
</div>
`;
